eitprocessing/datahandling/eitdata.py

```diff
@@ -69,16 +71,17 @@ def concatenate(self: T, other: T, newlabel: str | None = None) -> T:  # noqa: D
         # Check that data can be concatenated
         self.isequivalent(other, raise_=True)
         if np.min(other.time) <= np.max(self.time):
-            msg = f"{other} (b) starts before {self} (a) ends."
+            msg = f"Concatenation a+b failed because {other.name} (b) starts before {self.name} (a) ends."
```

Comment:
I like the clarity, but not the fact a+b is in the message. Maybe something like:

```suggestion
            msg = f"Concatenation failed because {other.name} (second dataset) starts before {self.name} (first dataset) ends."
```

Reply:
I made it 
```python
f"Concatenation failed. Second dataset ({other.name}) may not start before first ({self.name}) ends."
```
so that it fits on one line